File: clawpack-4.5.0/apps/advection/2d/annulus/mapc2p.py


def mapc2p(xc,yc):
    """
    Specifies the mapping to curvilinear coordinates -- should be consistent
    with mapc2p.f
    """
    from numpy import sin, cos

    # Polar coordinates (xc = r,  yc = theta)
    xp = xc * cos(yc)
    yp = xc * sin(yc)
    return xp,yp

File: clawpack-4.5.0/apps/advection/2d/annulus/mapc2p.f
c
c     =====================================================
      subroutine mapc2p(xc,yc,xp,yp)
c     =====================================================
c
c     # on input,  (xc,yc) is a computational grid point
c     # on output, (xp,yp) is corresponding point in physical space
c
      implicit double precision (a-h,o-z)
c
c     # Polar coordinates, xc = r,  yc = theta
      xp = xc * dcos(yc) 
      yp = xc * dsin(yc)
      return
      end
